src/infrastructure/note.repository.ts

```diff
@@ -1,11 +1,37 @@
 import NoteRepositoryInterface from '../domain/note.repository.interface';
 import Note from '../domain/entities/Note';
 import NoteStorage from './storage/note';
+import Transport from './transport';
+
+/**
+ * Get note response
+ */
+interface GetNoteResponse {
```

Comment:
also, I'd suggest not to duplicate properties here:

```ts
type GetNoteResponsePayload = Note;
```

I'm using "Paylaod" because API can have some format of all responses like:


```ts
interface ApiSuccessfulResponse<Payload> {
  payload: Payload;
}

interface ApiErrorResponse {
  error: number; // 500, 404, etc
  code: number // custom error code means error cause. Will be used to display a message to user. Message will be got from i18n dictionary by this code.
}

type ApiResponse<Payload> = ApiSuccessfulResponse<Payload> | ApiErrorResponse;
 
```

Only `NotesApiTransport` should know about API messages format. Repostory should got a `payload`

Reply:
done